nmigen/hdl/dsl.py

```diff
@@ -124,7 +130,8 @@ def __init__(self):
         self._ctrl_stack   = []
 
         self._driving      = SignalDict()
-        self._submodules   = []
+        self._submodules   = {}
```

Comment:
Nit: should this be something like `self._named_submodules`?

Reply:
Sure, let me change that